appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/VulnerabilityStore.java

```diff
@@ -0,0 +1,69 @@
+/*-
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full license.
+ */
+
+package com.vaadin.appsec.backend;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.cyclonedx.model.Component;
+
+import com.vaadin.appsec.backend.model.osv.response.OpenSourceVulnerability;
+
+/**
+ * Provides means to store and fetch vulnerabilities from a static instance.
+ */
+class VulnerabilityStore {
+
+    private final List<OpenSourceVulnerability> vulnerabilities = new ArrayList<>();
+
+    private final OpenSourceVulnerabilityService osvService;
+
+    private final BillOfMaterialsStore bomStore;
+
+    VulnerabilityStore(OpenSourceVulnerabilityService osvService,
+            BillOfMaterialsStore bomStore) {
+        this.osvService = osvService;
+        this.bomStore = bomStore;
+    }
+
+    /**
+     * Initializes the store with the given vulnerabilities.
+     *
+     * @param vulnerabilities
+     *            list of vulnerabilities to store
+     */
+    void init(List<OpenSourceVulnerability> vulnerabilities) {
+        this.vulnerabilities.clear();
+        this.vulnerabilities.addAll(vulnerabilities);
+    }
+
+    /**
+     * Returns all stored vulnerabilities from the store.
+     *
+     * @return list of stored vulnerabilities
+     */
+    List<OpenSourceVulnerability> getVulnerabilities() {
+        return Collections.unmodifiableList(vulnerabilities);
+    }
+
+    /**
+     * Refreshes vulnerability store from database.
+     *
+     * @return updated list of vulnerabilities
+     */
+    List<OpenSourceVulnerability> refresh() {
+        List<Component> components = bomStore.getBom().getComponents();
```

Comment:
When we do the refresh and we get the BOM again do we want to read the BOM file again with the `BillOfMaterialsStore.readBomFile()` at this point? My thinking is that the dependencies can change during one day and if the user regenerates the BOM file then we should use the new one instead of the one stored in the memory. What do you think?

Reply:
I think the BOM can't be regenerated while the app is running. And if it's restarted it will be read again anyway.

Reply:
Yes, you are right. I didn't think this through properly.